sensor/AdblockPlugin.js

```diff
@@ -130,38 +130,41 @@ class AdblockPlugin extends Sensor {
     async applySystemAdblock() {
         if (this.systemSwitch && this.adminSystemSwitch) {
             const adblocktagset = `mac-address-tag=%${systemLevelMac}${dnsTag}\n`;
-            await fs.writeFile(systemConfigFile, adblocktagset);
+            await fs.writeFileAsync(systemConfigFile, adblocktagset);
         } else {
-            await fs.unlink(systemConfigFile, err => {
-                if (err) {
-                    if (err.code === 'ENOENT') {
-                        log.info(`Dnsmasq: No ${systemConfigFile}, skip remove`);
-                    } else {
-                        log.warn(`Dnsmasq: Error when remove ${systemConfigFile}`, err);
-                    }
+            try {
+                await fs.unlinkAsync(systemConfigFile);
+            } catch (err) {
+                if (err.code === 'ENOENT') {
+                    log.info(`Dnsmasq: No ${systemConfigFile}, skip remove`);
+                } else {
+                    log.warn(`Dnsmasq: Error when remove ${systemConfigFile}`, err);
                 }
-            });
+            }
         }
         dnsmasq.controlFilter('adblock', this.adminSystemSwitch);
     }
 
     async applyDeviceAdblock() {
         const macAddressArr = this.enabledMacAddresses;
         if (macAddressArr.length > 0 && this.adminSystemSwitch) {
-            const adblocktagset = `mac-address-tag=%${macAddressArr.join("%")}${dnsTag}\n`;
-            await fs.writeFile(deviceConfigFile, adblocktagset);
+            let adblocktagset = "";
+            macAddressArr.forEach((macAddress) => {
```

Comment:
just FYI: do not use forEach when you are going to call async function in the inner loop of forEach. Here just a string concat is fine.

use 
```
for(const element of array) {
  await ...
}
```

because **forEach** is Pseudo concurrent execution, the async functions can't be "await"-ed by caller.